test/file-system.ts

```diff
@@ -0,0 +1,131 @@
+///<reference path=".d.ts"/>
+"use strict";
+
+import yok = require("../lib/common/yok");
+
+import Future = require("fibers/future");
+import stubs = require("./stubs");
+import path = require("path");
+import temp = require("temp");
+import util = require("util");
+import hostInfo = require("../lib/common/host-info");
+var assert = require("chai").assert;
+
+var fileSystemFile = require("../lib/common/file-system");
+var cordovaBlankTemplateZipFile = path.join(__dirname, "../resources/ProjectTemplates/Telerik.Mobile.Cordova.Blank.zip");
+var cordovaBlankTemplateZipFileIncorrectName = path.join(__dirname, "../resources/ProjectTemplates/Telerik.Mobile.Cordova.blank.zip");
+import childProcessLib = require("../lib/common/child-process");
+import staticConfigLib = require("../lib/config");
+var isOsCaseInsensitive = hostInfo.isLinux();
```

Comment:
This code is correct, but for the lulz: https://msdn.microsoft.com/en-us/library/windows/desktop/aa363858(v=vs.85).aspx and scroll down to FILE_FLAG_POSIX_SEMANTICS